src/Catalyst.Core.Modules.Dfs/CoreApi/BlockApi.cs

```diff
@@ -0,0 +1,396 @@
+                        await stream.WriteAsync(block.DataBytes, 0, block.DataBytes.Length, cancel)
+                           .ConfigureAwait(false);
+                    },
+
+                    Deserialize = async (stream, cid, cancel) =>
+                    {
+                        var block = new DataBlock
+                        {
+                            Id = cid,
+                            Size = stream.Length
+                        };
+
+                        block.DataBytes = new byte[block.Size];
+
+                        for (int i = 0, n; i < block.Size; i += n)
+                        {
+                            n = await stream.ReadAsync(block.DataBytes, i, (int) block.Size - i, cancel)
+                               .ConfigureAwait(false);
+                        }
+
+                        return block;
+                    }
+                };
+
+                return _store;
+            }
+        }
+
+        public async Task<IDataBlock> GetAsync(Cid id, CancellationToken cancel = default)
+        {
+            // Hack for empty object and empty directory object
+            var emptyDirectory = GetEmptyDirectory();
+            if (id == emptyDirectory.Id)
+            {
+                return emptyDirectory;
+            }
+
+            var emptyNode = GetEmptyNode();
+            if (id == emptyNode.Id)
+            {
+                return emptyNode;
+            }
+
+            // If identity hash, then CID has the content.
+            if (id.Hash.IsIdentityHash)
+            {
+                return new DataBlock
+                {
+                    DataBytes = id.Hash.Digest,
+                    Id = id,
+                    Size = id.Hash.Digest.Length
+                };
+            }
+
+            // Check the local filesystem for the block.
+            var block = await Store.TryGetAsync(id, cancel).ConfigureAwait(false);
+
+            if (block != null)
+            {
+                return block;
+            }
+
+            // Query the network, via DHT, for peers that can provide the
+            // content.  As a provider peer is found, it is connected to and
+            // the bitSwap want lists are exchanged.  Hopefully the provider will
+            // then send the block to us via bitSwap and the get task will finish.
+            using (var queryCancel = CancellationTokenSource.CreateLinkedTokenSource(cancel))
+            {
+                var bitswapGet = await _bitSwapApi.GetAsync(id, queryCancel.Token).ConfigureAwait(false);
+                var _ = _dhtApi.FindProvidersAsync(
+                    id,
+                    20, // TODO: remove this
+                    peer =>
+                    {
+                        var __ = ProviderFoundAsync(peer, queryCancel.Token).ConfigureAwait(false);
+                    },
+                    queryCancel.Token
+                );
+
+                //log.Debug("bitswap got the block");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)